src/CommandingOfficers/RuinedCommander.java

```diff
@@ -0,0 +1,437 @@
+package CommandingOfficers;
+
+import Engine.Army;
+import Engine.GameAction;
+import Engine.GameActionSet;
+import Engine.GameInstance;
+import Engine.GamePath;
+import Engine.GameScenario;
+import Engine.UnitActionFactory;
+import Engine.Utils;
+import Engine.XYCoord;
+
+import java.awt.Color;
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+
+import CommandingOfficers.CommanderInfo.InfoPage;
+import Engine.Combat.StrikeParams;
+import Engine.Combat.StrikeParams.BattleParams;
+import Engine.GameEvents.CommanderAbilityEvent;
+import Engine.GameEvents.GameEventQueue;
+import Engine.StateTrackers.KillCountsTracker;
+import Engine.StateTrackers.StateTracker;
+import Engine.UnitMods.UnitModifierWithDefaults;
+import Terrain.GameMap;
+import Terrain.MapMaster;
+import UI.GameOverlay;
+import Units.Unit;
+import Units.UnitContext;
+import Units.UnitDelta;
+
+public abstract class RuinedCommander extends DeployableCommander
+{
+  private static final long serialVersionUID = 1L;
+  public static final InfoPage DOR_MECHANICS_BLURB = new InfoPage(
+            "Days of Ruin mechanics:\n"
+          + "Defense is much weaker; Damage is base x attack/defense, vs base x attack x (200-defense)\n"
+          + "You have a CO zone that covers the area around your COU, and grants bonuses to your units inside.\n"
+          + "The zone expands 1 tile when your meter is half full, and again at full charge.\n"
+          + "All units get +10 attack and +10 defense while in the zone.\n"
+          + "Your CO may also have another zone boost that stacks with that boost, but may only apply to some unit types.\n"
+          + "The zone becomes global while your power is active, but activating your power requires your COU's action.\n"
+          + "Zone boosts apply to other COs in a tag, even when tagged out.\n"
+          + "CO Energy gained reflects HP damage you deal, and not on the type of unit you hit.\n"
+          + "You can only gain CO energy for combat where the initiator is inside the zone.\n"
+          + "  (This means an enemy in your zone attacking outside your zone can give you charge from your counterattack)\n"
+          + "However, your CO ability does not increase in cost as you use it.\n"
+          + "Veterancy:\n"
+          + "On making a kill, your unit will level up.\n"
+          + "There are 3 veterancy levels; their stat boosts are +5/0, +10/0, and +20/20 attack/defense.\n"
+          + "Your COU is always max level.\n"
+          + "Veterancy bonuses do not apply if this CO is tagged out, but kills are always tracked.\n"
+          );
+  public static final int DOR_STAR_VALUE = 10000;
+  public static final int CHARGERATIO_HP =  1000; // Funds value of 1 HP damage dealt, for the purpose of power charge
```

Comment:
This doesn't vary based on the unit? Or wait, that changed between the OT and DoR, didn't it; they used destroyed funds, and DoR used straight HP?

Reply:
Correct.
DoR does straight HP damage *dealt* only, and only if the combat initiator is inside your zone (rather than, y'know, your unit).